packages/office-ui-fabric-react/src/utilities/grid/GridCell.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import {
+  autobind,
+  BaseComponent,
+  css,
+  getId,
+} from '../../Utilities';
+import { IGridCellProps } from './GridCell.Props';
+import { CommandButton } from '../../Button';
+
+export class GridCell<T, P extends IGridCellProps<T>> extends React.Component<P, {}> {
+
+  public static defaultProps = {
+    cellShape: 'circle',
+    disabled: false,
+    id: getId('gridCell')
+  };
+
+  private _id: string;
+
+  constructor(props?: P) {
+    super();
+  }
+
+  public render() {
+    let {
+      item,
+      id,
+      className,
+      role,
+      selected,
+      disabled,
+      onClick,
+      onHover,
+      onFocus,
+      onRenderItem,
+      cellDisabledStyle,
+      cellIsSelectedStyle,
+      index,
+      label
+    } = this.props;
+    return (
+      <CommandButton
+        id={ id + '-item' + index }
+        data-index={ index }
+        data-is-focusable={ true }
+        disabled={ disabled }
+        className={ css(className,
+          {
+            ['' + cellIsSelectedStyle]: selected,
+            ['' + cellDisabledStyle]: disabled
+          }
+        ) }
+        onClick={ this._onClick }
+        onMouseEnter={ this._onMouseEnter }
+        onMouseLeave={ this._onMouseLeave }
+        onFocus={ this._onFocus }
+        role={ role }
+        aria-selected={ selected }
+        ariaLabel={ label }
+        title={ label }
+      >
+        { onRenderItem(item) }
+      </CommandButton >
+    );
+  }
+
+  @autobind
+  private _onClick() {
+    let {
+        onClick,
+      disabled,
+      item
+      } = this.props as P;
```

Comment:
I get the following error if I don't.
Cannot invoke an expression whose type lacks a call signature. Type '((item: T) => void) | undefined' has no compatible call signatures.

I believe this is an issue with the callback being nullable. You can see I don't have to do this in the render() function. Looking around in the rest of our codebase were we do generics and have nullable functions we do the explicit cast which seems to fix the issue. I would have assumed TS would be smart enough to know that this.props is a P but it might be a limitation or known issue with TypeScript.